workstation.tf

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 Chainguard, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+resource "google_compute_network" "default" {
+  provider                = google-beta
+  name                    = var.name
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "default" {
+  provider      = google-beta
+  name          = var.name
+  ip_cidr_range = "10.0.0.0/24"
+  region        = var.region
+  network       = google_compute_network.default.name
+}
+
+resource "google_workstations_workstation_cluster" "default" {
+  provider               = google-beta
+  workstation_cluster_id = var.name
+  network                = google_compute_network.default.id
+  subnetwork             = google_compute_subnetwork.default.id
+  location               = var.region
+}
+
+data "google_compute_default_service_account" "default" {}
+
+resource "google_workstations_workstation_config" "default" {
+  provider               = google-beta
+  workstation_config_id  = var.name
+  workstation_cluster_id = google_workstations_workstation_cluster.default.workstation_cluster_id
+  location               = var.region
+
+  idle_timeout    = var.idle_timeout
+  running_timeout = var.running_timeout
+
+  host {
+    gce_instance {
+      machine_type      = var.machine_type
+      boot_disk_size_gb = var.disk_gb
+      // TODO: make this true and figure out how to give it access to the image.
+      disable_public_ip_addresses = false
+      // TODO: Use a less permissive service account.
+      service_account = data.google_compute_default_service_account.default.email
```

Comment:
We should definitely do this, and frankly it is pointless to give it any credentials because they don't run it with `cloud-platform`, so you need to plumb auth another way.

Reply:
The service account needs access to pull the image if it's private (and I think it should be private), otherwise I'm not sure what else it needs. I tried creating a minimal SA with `roles/artifactregistry.reader` and `roles/viewer` and neither worked, and it takes foooorrrreeeeevvvver to iterate on this, so I'm sort of inclined to punt for now.

Reply:
Ah yeah, I was thinking that the GSA would be used for user auth, which is why I was disappointed about the scopes 😞